electrum/plugins/satochip/satochip.py

```diff
@@ -0,0 +1,651 @@
+    
+    def detect_smartcard_reader(self):
+        _logger.info(f"[SatochipPlugin] detect_smartcard_reader")#debugSatochip
+        self.cardtype = AnyCardType()
+        try:
+            cardrequest = CardRequest(timeout=5, cardType=self.cardtype)
+            cardservice = cardrequest.waitforcard()
+            return [Device(path="/satochip",
+                           interface_number=-1,
+                           id_="/satochip",
+                           product_key=(SATOCHIP_VID,SATOCHIP_PID),
+                           usage_page=0,
+                           transport_ui_string='ccid')]
+        except CardRequestTimeoutException:
+            _logger.info(f'time-out: no card inserted during last 5s')
+            return []
+        except Exception as exc:
+            _logger.info(f"Error during connection:{str(exc)}")
+            return []
+        return []
+        
+    
+    def create_client(self, device, handler):
+        _logger.info(f"[SatochipPlugin] create_client()")#debugSatochip
+        
+        if handler:
+            self.handler = handler
+
+        try:
+            rv = SatochipClient(self, handler)
+            return rv
+        except Exception as e:
+            _logger.exception(f"[SatochipPlugin] create_client() exception: {str(e)}")
+            return None
+
+    def setup_device(self, device_info, wizard, purpose):
+        _logger.info(f"[SatochipPlugin] setup_device()")#debugSatochip
+        
+        devmgr = self.device_manager()
+        device_id = device_info.device.id_
+        client = devmgr.client_by_id(device_id)
+        if client is None:
+            raise Exception(_('Failed to create a client for this device.') + '\n' +
+                            _('Make sure it is in the correct state.'))
+        client.handler = self.create_handler(wizard)
+        
+        # check applet version
+        while(True):
+            (response, sw1, sw2, d)=client.cc.card_get_status()
+            if (sw1==0x90 and sw2==0x00):
+                v_supported= (CardConnector.SATOCHIP_PROTOCOL_MAJOR_VERSION<<8)+CardConnector.SATOCHIP_PROTOCOL_MINOR_VERSION
+                v_applet= (d["protocol_major_version"]<<8)+d["protocol_minor_version"] 
+                strcmp= 'lower' if (v_applet<v_supported) else 'higher'   
+                _logger.info(f"[SatochipPlugin] setup_device(): Satochip version={hex(v_applet)} Electrum supported version= {hex(v_supported)}")#debugSatochip
+                if (v_supported!=v_applet):
+                    msg=_('The version of your Satochip (v{v_applet_maj:x}.{v_applet_min:x}) is {strcmp} than supported by Electrum (v{v_supported_maj:x}.{v_supported_min:x}). You should update Electrum to ensure correct function!').format(strcmp=strcmp, v_applet_maj=d["protocol_major_version"], v_applet_min=d["protocol_minor_version"],  v_supported_maj=CardConnector.SATOCHIP_PROTOCOL_MAJOR_VERSION, v_supported_min=CardConnector.SATOCHIP_PROTOCOL_MINOR_VERSION)
```

Comment:
`_()` translates/localizes strings
 maybe split this up, and remove the named parameters inside the string, as this seems really error-prone to translators
Or alternatively, do not translate it.